crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -1335,15 +1335,19 @@ NamedExpressionTest: ast::Expr = {
     Test<"all">,
 }
 
+NamedExpressionName: ast::Expr = {
+    <location:@L> <id:Identifier> <end_location:@R> => ast::Expr::Name(
+        ast::ExprName { id: id.into(), ctx: ast::ExprContext::Store, range: (location..end_location).into() },
+    ),
+}
+
 NamedExpression: ast::Expr = {
-    <location:@L> <id:Identifier> <end_location:@R> ":=" <value:Test<"all">> => {
+    <location:@L> <target:NamedExpressionName> ":=" <value:Test<"all">> <end_location:@R> => {
         ast::Expr::NamedExpr(
             ast::ExprNamedExpr {
-                target: Box::new(ast::Expr::Name(
-                    ast::ExprName { id: id.into(), ctx: ast::ExprContext::Store, range: (location..end_location).into() },
-                )),
-                range: (location..value.end()).into(),
```

Comment:
In the example from the summary, `value.end()` is the end of `y`, not the end of its parenthesized range.